diff --git a/includes/entity.inc b/includes/entity.inc
index 2fefd59..594eafd 100644
--- a/includes/entity.inc
+++ b/includes/entity.inc
@@ -175,7 +175,7 @@ class DrupalDefaultEntityController implements DrupalEntityControllerInterface {
     // were passed. The $ids array is reduced as items are loaded from cache,
     // and we need to know if it's empty for this reason to avoid querying the
     // database when all requested entities are loaded from cache.
-    $passed_ids = !empty($ids) ? array_flip($ids) : FALSE;
+    $passed_ids = !empty($ids) ? array_flip(array_filter($ids, 'strlen')) : FALSE;
     // Try to load entities from the static cache, if the entity type supports
     // static caching.
     if ($this->cache && !$revision_id) {
@@ -352,7 +352,7 @@ class DrupalDefaultEntityController implements DrupalEntityControllerInterface {
     // Load any available entities from the internal cache.
     if (!empty($this->entityCache)) {
       if ($ids) {
-        $entities += array_intersect_key($this->entityCache, array_flip($ids));
+        $entities += array_intersect_key($this->entityCache, array_flip(array_filter($ids, 'strlen')));
       }
       // If loading entities only by conditions, fetch all available entities
       // from the cache. Entities which don't match are removed later.
